app/services/CoreVatReturnService.scala

```diff
@@ -158,28 +158,44 @@ class CoreVatReturnService @Inject()(
   }
 
   private def getEuTraderIdForCountry(country: Country, registration: Registration): Option[CoreEuTraderId] = {
-    val matchedRegistration = registration.euRegistrations.filter {
+    val matchedRegistration = registration.euRegistrations.filter
+    {
       case euRegistration: EuVatRegistration => euRegistration.country == country
       case euRegistrationWithFE: RegistrationWithFixedEstablishment => euRegistrationWithFE.country == country
       case euRegistrationWithoutTaxId: RegistrationWithoutTaxId => euRegistrationWithoutTaxId.country == country
       case euRegistrationWithoutFE: RegistrationWithoutFixedEstablishment => euRegistrationWithoutFE.country == country
     }
 
     matchedRegistration.headOption.flatMap {
-      case euRegistration: EuVatRegistration => Some(CoreEuTraderVatId(euRegistration.vatNumber, euRegistration.country.code))
       case euRegistrationWithFETaxId: RegistrationWithFixedEstablishment =>
-        if(euRegistrationWithFETaxId.taxIdentifier.identifierType.equals(Vat)) {
-          Some(CoreEuTraderVatId(euRegistrationWithFETaxId.taxIdentifier.value, euRegistrationWithFETaxId.country.code))
+        logger.info("sending tax id for fixed establishment")
+        val taxIdValue =
+          if(euRegistrationWithFETaxId.taxIdentifier.value.startsWith(euRegistrationWithFETaxId.country.code)) {
+            country.code match {
+              case "FR" => if(euRegistrationWithFETaxId.taxIdentifier.value.length == 13) {
+                logger.info(s"Stripping country code for ${euRegistrationWithFETaxId.country.name}")
+                euRegistrationWithFETaxId.taxIdentifier.value.substring(euRegistrationWithFETaxId.country.code.length)
```

Comment:
It's a little dense to read - could maybe extract euRegistrationWithFETaxId.taxIdentifier.value and euRegistrationWithFETaxId.country to variables?